blockchain/network.py

```diff
@@ -473,14 +549,54 @@ def resolve_data(self, node, guids, phase):
             try:
                 verification = thrift_converter.convert_thrift_verification(verification)
                 verification_db.insert_verification(verification, verification['verification_id'])
+                # check if there are nodes further down the chain interested in this record
                 replicated_verifications = verification_db.get_all_replication(verification['block_id'], phase, verification['origin_id'])
                 for replicated_ver in replicated_verifications:
                     vr_transfers_db.insert_transfer(replicated_ver['origin_id'], replicated_ver['signature']['signatory'], verification['verification_id'])
+                # check if there are active subscriptions interested in this record
+                self.update_subscription_response(verification)
             except Exception as ex:
                 template = "An exception of type {0} occured. Arguments:\n{1!r}"
                 message = template.format(type(ex).__name__, ex.args)
                 logger().warning(message)
 
+    def update_subscription_response(self, verification_record):
+        """ check sub_vr_backlog, check if there are any active subscriptions that have phase criteria
+            that match given vr phase. if so, build a response of transactions and matching vrs for this block. """
+        block_id = verification_record['block_id']
+        backlogs = backlog_db.get_backlogs(block_id)
+        if backlogs:
```

Comment:
We will need to consider de-duplication on the `sub_vr_transfers` or on the client.